src/commands/deployWorkspaceProject/deploymentConfiguration/workspace/TryUseExistingWorkspaceRegistryStep.ts

```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { type Registry } from "@azure/arm-containerregistry";
+import { GenericTreeItem, activitySuccessContext, activitySuccessIcon, nonNullProp } from "@microsoft/vscode-azext-utils";
+import { type Progress } from "vscode";
+import { ExecuteActivityOutputStepBase, type ExecuteActivityOutput } from "../../../../utils/activity/ExecuteActivityOutputStepBase";
+import { createActivityChildContext } from "../../../../utils/activity/activityUtils";
+import { localize } from "../../../../utils/localize";
+import { AcrListStep } from "../../../image/imageSource/containerRegistry/acr/AcrListStep";
+import { type DeploymentConfigurationSettings } from "../../settings/DeployWorkspaceProjectSettingsV2";
+import { dwpSettingUtilsV2 } from "../../settings/dwpSettingUtilsV2";
+import { type WorkspaceDeploymentConfigurationContext } from "./WorkspaceDeploymentConfigurationContext";
+
+export class TryUseExistingWorkspaceRegistryStep extends ExecuteActivityOutputStepBase<WorkspaceDeploymentConfigurationContext> {
+    public priority: number = 220;  /** Todo: Figure out a good priority level */
+
+    protected async executeCore(context: WorkspaceDeploymentConfigurationContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        this.options.shouldSwallowError = true;
+
+        const settings: DeploymentConfigurationSettings[] | undefined = await dwpSettingUtilsV2.getWorkspaceDeploymentConfigurations(nonNullProp(context, 'rootFolder'));
+        if (!settings?.length) {
+            return;
+        }
+
+        progress.report({ message: localize('searchingAvailableRegistries', 'Searching for available registry...') });
+        const registries: Registry[] = await AcrListStep.getRegistries(context);
+        const registryMap: Map<string, Registry> = new Map();
```

Comment:
Used a map here to reduce operations to O(n).  Using a `.find` inside of a for loop would've been ~O(n^2).  Not sure if we really care here though 🤷‍♂️

Reply:
Yeah but that O(n) space you're allocating now 😮 

Haha, I don't have a strong preference either way. I think it makes sense to use a map here.

Reply:
Changing the logic quite a bit so no longer relevant as much, though still using a map :D